kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/default-language-form/index.vue

```diff
@@ -85,55 +37,9 @@
         required: false,
       },
     },
-    data() {
-      return {
-        selectedLanguageId: this.currentLanguageId,
-      };
-    },
-    computed: {
-      selectedLanguage() {
-        return allLanguages[this.selectedLanguageId].lang_name;
-      },
-      remainingLanguages() {
-        const remainingLanguages = Object.values(omit(allLanguages, [this.selectedLanguageId]));
-        remainingLanguages.sort((lang1, lang2) => {
-          // puts words with foreign characters first in the array
-          return lang2.lang_name.localeCompare(lang1.lang_name);
-        });
-
-        return remainingLanguages;
-      },
-      numberOfLanguageButtons() {
-        let numBtns = this.isMobile ? mobileNumberOfLanguageButtons : desktopNumberOfLanguageButtons;
-        // prevent a case where the selector menu has just a single item
-        if (Object.keys(allLanguages).length === numBtns + 2) {
-          return numBtns + 1;
-        }
-        return numBtns;
-      },
-      buttonLanguages() {
-        return this.remainingLanguages.slice(0, this.numberOfLanguageButtons);
-      },
-      selectorLanguages() {
-        return this.remainingLanguages.slice(this.numberOfLanguageButtons);
-      },
-    },
     methods: {
-      changeLanguage(languageId) {
-        this.selectedLanguageId = languageId;
-        const path = this.Kolibri.urls['kolibri:set_language']();
-        const entity = {
-          language: this.selectedLanguageId,
-        };
-        httpClient({
-          path,
-          entity,
-        }).then(() => {
-          global.location.reload(true);
-        });
-      },
       setLanguage() {
-        this.submitDefaultLanguage(this.selectedLanguageId);
+        this.submitDefaultLanguage(this.currentLanguageId);
```

Comment:
Where is currentLanguageId defined? I don't see it anywhere here, or any communication between the language list component and this component?

Reply:
it's defined below as a vuex getter: https://github.com/learningequality/kolibri/pull/2259/files/20b9bdadc517763cb036fadca183c7989e5a7bf3#diff-4ff2174816eb01fe1a5db7055c66e187R51

(was hidden in github's default view of the diff)